tracker/server.go

```diff
@@ -0,0 +1,237 @@
+package tracker
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/R-a-dio/valkyrie/config"
+	"github.com/R-a-dio/valkyrie/errors"
+	"github.com/R-a-dio/valkyrie/proxy/compat"
+	"github.com/R-a-dio/valkyrie/rpc"
+	"github.com/R-a-dio/valkyrie/util/graceful"
+	"github.com/R-a-dio/valkyrie/website"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/hlog"
+	"google.golang.org/grpc"
+)
+
+type Server struct {
+	cfg     config.Config
+	mu      sync.Mutex
+	rpcSrv  *grpc.Server
+	rpcLn   net.Listener
+	httpSrv *http.Server
+	httpLn  net.Listener
+}
+
+func NewGRPCServer(ctx context.Context, lts radio.ListenerTrackerService) *grpc.Server {
+	gs := rpc.NewGrpcServer(ctx)
+	rpc.RegisterListenerTrackerServer(gs, rpc.NewListenerTracker(lts))
+	return gs
+}
+
+func NewServer(ctx context.Context, cfg config.Config) (*Server, error) {
+	const op errors.Op = "tracker.NewServer"
+
+	// setup recorder
+	var recorder = NewRecorder(ctx, cfg)
+
+	// setup periodic task to update the manager of our listener count
+	go PeriodicallyUpdateListeners(ctx, cfg.Manager, recorder, UpdateListenersTickrate)
+	// setup periodic task to keep recorder state in sync with icecast
+	go PeriodicallySyncListeners(ctx, cfg, recorder, SyncListenersTickrate)
```

Comment:
this whole setup should probably either be moved to `Start` or kept outside since we don't want to actually run any logic until the server is ready to actually serve (which it only is after handling the graceful stuff and calling `Start`)

You can use the `graceful.Sync` channel inside of the periodic functions too as an alternative